src/sentry/web/frontend/auth_login.py

```diff
@@ -95,39 +243,238 @@ def get_register_form(self, request: Request, initial=None):
             auto_id="id_registration_%s",
         )
 
-    def can_register(self, request: Request):
-        return bool(has_user_registration() or request.session.get("can_register"))
+    def redirect_to_next_register_step(
+        self, request: Request, register_form: RegistrationForm, organization: Organization
+    ) -> HttpResponseRedirect:
+        """Given a valid register form, redirects the user to their next step."""
 
-    def get_join_request_link(self, organization):
-        if not organization:
-            return None
+        user = register_form.save()
+        user.send_confirm_emails(is_new_user=True)
+        user_signup.send_robust(sender=self, user=user, source="register-form", referrer="in-app")
 
-        if organization.get_option("sentry:join_requests") is False:
-            return None
+        # HACK: grab whatever the first backend is and assume it works
+        user.backend = settings.AUTHENTICATION_BACKENDS[0]
+        self._handle_login(request=request, user=user, organization=organization)
 
-        join_request_link_viewed.send_robust(sender=self, organization=organization)
+        # can_register should only allow a single registration
+        request.session.pop("can_register", None)
+        request.session.pop("invite_email", None)
 
-        return reverse("sentry-join-request", args=[organization.slug])
+        # Attempt to directly accept any pending invites
```

Comment:
All invite logic why don't we pull out into its own thing separate from a generic "redirec to next step" maybe?